addons/hr_expense/models/account_move.py

```diff
@@ -97,3 +97,17 @@ def _prepare_product_base_line_for_taxes_computation(self, product_line):
         if product_line.expense_id:
             results['special_mode'] = 'total_included'
         return results
+
+    def _reverse_moves(self, default_values_list=None, cancel=False):
+        # EXTENDS account
+        self.filtered('expense_ids').write({'expense_ids': [Command.clear()], 'ref': False})
+        # Ref also or when restarting the expense flow, we get duplicate issue on vendor.bill
```

Comment:
mmh what about changing the unicity constraint to ignore the reversed entries? Sounds like a better solution, no? :thinking:

Reply:
I don't think it's a good idea.
That makes the computation of the state way more complex, re-invoicing and payroll flows that is only giving a small benefit in terms of clarity. You already have a link back to the expense in the chatter of the reversed move to keep a historical link. We plan on adding a same kind of link from the expense to the move in the future, but not in this task

Reply:
Discussed with PO, we'll just keep the ref, the warning is non-blocking, only when the bill is reset to draft, and a nice to have in the case where mistakes were made